release/release-offline.ts

```diff
@@ -0,0 +1,410 @@
+/**
+ * Note: This is a backup and failsafe if you cannot use the release github actions in CI for some reason.
+ * The environment consistency and observability of the release process is very important, so
+ * you should only use this script if absolutely necessary, and you cannot use CI for some very
+ * good reason.
+ */
+
+import "dotenv/config";
+import { Octokit } from "@octokit/rest";
+import "zx/globals";
+import { $ } from "zx";
+$.verbose = false;
+
+import {
+  isValidVersionString,
+  hasBeenReleased,
+  isValidCommitHash,
+  isEnterpriseVersion,
+  getMajorVersion,
+  isLatestRelease,
+  getVersionInfo,
+  publishRelease,
+  closeMilestone,
+  openNextMilestones,
+  versionRequirements,
+} from "./src";
+
+const {
+  GITHUB_TOKEN,
+  GITHUB_OWNER,
+  GITHUB_REPO,
+  AWS_S3_DOWNLOADS_BUCKET,
+  AWS_S3_STATIC_BUCKET,
+  AWS_CLOUDFRONT_DOWNLOADS_ID,
+  AWS_CLOUDFRONT_STATIC_ID,
+  AWS_S3_RELEASE_SECRET_ACCESS_KEY,
+  AWS_S3_RELEASE_ACCESS_KEY_ID,
+  DOCKERHUB_OWNER,
+  DOCKERHUB_RELEASE_USERNAME,
+  DOCKERHUB_RELEASE_TOKEN,
+} = process.env;
+
+const github = new Octokit({ auth: GITHUB_TOKEN });
+
+const JAR_PATH = "../target/uberjar";
+
+const version = process.argv?.[2]?.trim();
+const commit = process.argv?.[3]?.trim();
```

Comment:
To match the `isValidCommitHash` argument name.

```suggestion
const commitHash = process.argv?.[3]?.trim();
```